scripts/gc.js

```diff
@@ -0,0 +1,433 @@
+  ]);
+
+   const els = [];
+   const elt = [];
+   let numKeys = 0;
+   for await (const data of asyncChunks(pipeline)) {
+     if (numKeys < maxKeys) {
+       const key = ethUtil.keccak256(Buffer.from(data.key, 'hex'));
+       const val = (ethereumAccountToRlp(data.value));
+       els.push({'key': key, 'val': val});
+       numKeys += 1;
+     }
+   }
+  return els;
+}
+
+const test_memdiff_ETH = async (els) => {
+  let trees = [];
+  
+  const maxBatches = process.argv[4];
+  
+  let state = new ethereumjs();
+
+  console.log("ETH Tree addition");
+  console.log("Batch, ", "memdiff(kB), ", "memusage(MB), ", "memtotal(MB)");
+
+  let memi = process.memoryUsage().heapUsed;
+  let memf, memdiff;
+  for (let i = 0; i < maxBatches; i++) {
+       
+    let flag_ETH = false;
+    state.batch(els, () => {flag_ETH = true});
+    wait.for.predicate(() => flag_ETH);
+    trees.push(state);
+    state = new ethereumjs();
+
+    memf = process.memoryUsage().heapUsed;
+    memdiff = memf - memi;
+    memi = memf;
+    console.log(`${i}, ${(memdiff/1024).toFixed(2)}, ${(memf/(1024*1024)).toFixed(2)}, ${(process.memoryUsage().heapTotal/(1024*1024)).toFixed(2)}`);
+  }
+
+  console.log("ETH Tree removal");
+  console.log("Batch, ", "memdiff(kB), ", "memusage(MB), ", "memtotal(MB)");
+  
+  memi = process.memoryUsage().heapUsed;
+  for (let i = 0; i < maxBatches; i++) {
+    trees.shift();
+    global.gc();
+    memf = process.memoryUsage().heapUsed;
+    memdiff = memf-memi;
+    memi = memf;
+    console.log(`${i}, ${(memdiff/1024).toFixed(2)}, ${(memf/(1024*1024)).toFixed(2)}, ${(process.memoryUsage().heapTotal/(1024*1024)).toFixed(2)}`);
+  }
+}
+const test_memdiff_RBC = async (els) => {
+  let trees = [];
+  
+  const maxBatches = process.argv[4];
+  
+  let state = new rainblock.MerklePatriciaTree();
+
+  console.log("Tree addition");
+  console.log("Batch, ", "memdiff(kB), ", "memusage(MB), ", "memtotal(MB)");
+
+  let memi = process.memoryUsage().heapUsed;
+  let memf, memdiff;
+  for (let i = 0; i < maxBatches; i++) {
+    if (trees.length) {
+      const prevTree = trees.pop();
+      trees.push(prevTree);
+      trees.push(prevTree.batchCOW(els));
+    } else {
+      trees.push(state.batchCOW(els));
+    }
+    memf = process.memoryUsage().heapUsed;
+    memdiff = memf - memi;
+    memi = memf;
+    console.log(`${i}, ${(memdiff/1024).toFixed(2)}, ${(memf/(1024*1024)).toFixed(2)}, ${(process.memoryUsage().heapTotal/(1024*1024)).toFixed(2)}`);
+  }
+
+  console.log("Tree removal");
+  console.log("Batch, ", "memdiff(kB), ", "memusage(MB), ", "memtotal(MB)");
+  
+  memi = process.memoryUsage().heapUsed;
+  for (let i = 0; i < maxBatches; i++) {
+    trees.shift();
+    global.gc();
+    memf = process.memoryUsage().heapUsed;
+    memdiff = memf-memi;
+    memi = memf;
+    console.log(`${i}, ${(memdiff/1024).toFixed(2)}, ${(memf/(1024*1024)).toFixed(2)}, ${(process.memoryUsage().heapTotal/(1024*1024)).toFixed(2)}`);
+  }
+}
+
```

Comment:
Could we have one function which takes state as input and measures the memory difference?
testMemDiff( state )?